File: src/main/java/um/haberes/report/model/haberes/core/BonoImpresionDto.java
package um.haberes.report.model.haberes.core;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.OffsetDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BonoImpresionDto {

    private Long bonoImpresionId;

    private Long legajoId;

    private int anho = 0;

    private int mes = 0;

    private Long legajoIdSolicitud;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fecha;

    private String ipAddress = "";

}

File: src/main/java/um/haberes/report/kotlin/dto/haberes/core/BonoImpresionDto.kt
package um.haberes.report.kotlin.dto.haberes.core

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.OffsetDateTime

data class BonoImpresionDto(

    var bonoImpresionId: Long? = null,
    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var legajoIdSolicitud: Long? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fecha: OffsetDateTime? = null,
    var ipAddress: String = ""

)
